roboflow/core/version.py

```diff
@@ -85,54 +89,77 @@ def __init__(self, version_dict, type, api_key, name, version, model_format, loc
                 self.model = None
 
     def download(self, model_format=None, location=None):
+        """
+        Download and extract a ZIP of a version's dataset in a given format
+
+        :param model_format: A format to use for downloading
+        :param location: An optional path for saving the file
 
+        :return: Dataset
+        """
         if location is None:
-            if "DATASET_DIRECTORY" in os.environ:
-                location = (
-                    os.environ["DATASET_DIRECTORY"]
-                    + "/"
-                    + self.name.replace(" ", "-")
-                    + "-"
-                    + self.version
-                )
-            else:
-                location = self.name.replace(" ", "-") + "-" + self.version
+            location = self.__get_download_location()
```

Comment:
this same URL is helpful for extracting to, so i popped it into a private helper.

Reply:
Out of curiosity, do you know why `__` are used all around?

Reply:
good question!

this is a pattern that i was following in the library where "private" class methods seemed to be prefixed by a double underscore.

usually, in python a single underscore is used to define a private method, but either work. there's slightly more protection with the double underscore because outside of `self`, the method name is mangled a bit and harder to call directly.

Reply:
Thanks, the double `__` looks weird imho, no strong feelings as well